pandas/core/internals.py

```diff
@@ -1847,8 +1847,9 @@ def _can_hold_element(self, element):
         if tipo is not None:
             return (issubclass(tipo.type, (np.floating, np.integer)) and
                     not issubclass(tipo.type, (np.datetime64, np.timedelta64)))
-        return (isinstance(element, (float, int, np.floating, np.int_)) and
-                not isinstance(element, (bool, np.bool_, datetime, timedelta,
+        return (
+            isinstance(element, (float, int, np.floating, np.int_, compat.long))
```

Comment:
there is another section where you need to add this (in _can_hold_element), in ComplexBlock, also can you add a test which hits that as well